x-pack/platform/plugins/shared/observability_ai_assistant/server/functions/elasticsearch.ts

```diff
@@ -40,6 +52,18 @@ export function registerElasticsearchFunction({
       },
     },
     async ({ arguments: { method, path, body } }) => {
+      // Allowlist: (1) all GET requests, (2) POST requests whose *final* path segment is exactly "_search".
+      const [pathWithoutQuery] = path.split('?');
+      const pathSegments = pathWithoutQuery.replace(/^\//, '').split('/');
+      const lastPathSegment = pathSegments[pathSegments.length - 1];
+      const isSearchEndpoint = lastPathSegment === '_search';
+
+      if (method !== 'GET' && !(method === 'POST' && isSearchEndpoint)) {
```

Comment:
Can you extract this to a well-named helper (eg `isOperationAllowed`) to contain this logic.